state/machine.go

```diff
@@ -217,42 +218,20 @@ type instanceData struct {
 // upgradeSeriesLock holds the attributes relevant to lock a machine during a
 // series update of a machine
 type upgradeSeriesLock struct {
-	Id             string                     `bson:"machineid"`
-	ToSeries       string                     `bson:"toseries"`
-	FromSeries     string                     `bson:"fromseries"`
-	PrepareStatus  MachineSeriesUpgradeStatus `bson:"preparestatus"`
-	PrepareUnits   []unitStatus               `bson:"prepareunits"`
-	CompleteStatus MachineSeriesUpgradeStatus `bson:"completestatus"`
-	CompleteUnits  []unitStatus               `bson:"completeunits"`
+	Id             string                           `bson:"machineid"`
+	ToSeries       string                           `bson:"toseries"`
+	FromSeries     string                           `bson:"fromseries"`
+	PrepareStatus  model.MachineSeriesUpgradeStatus `bson:"preparestatus"`
```

Comment:
I like the idea of timestamps, and its something we've done for other status fields.
I'm the one that asked for a slice instead of a map, but I can admit fault in that the complexity around dealing with a slice doesn't outweight the benefit of not putting arbitrary data as BSON attribute names.

Reply:
I can add a Trello Card for adding a time stamp. I certainly do agree. However the PR is getting quite long in the tooth and scope creep is starting to set in. This PR was initially meant to be really tiny - it was meant to detect a lock has been created and fire the hook. I feel that much of the small errors that we are seeing are due to the size of the PR coupled with the rapid pace of development.